### PR TITLE
fix: access generator stepping outside analysis interval

### PR DESCRIPTION
**Pull request by vishwa2710**
Fixes a bug where if a solved coarse access interval is between the analysis interval boundary and the last step, then it steps outside the analysis interval.

For example, if the analysis interval ends at 13:30:25, and the step size is 1 minute, and the analysis interval ends at 13:30:00,
then if the last step satisfies the access condition at 13:29:25, it will try and check for the step beyond by 1 minute to see if it can find a precise crossing, even though it is outside the analysis interval.

To fix this, we only solve for the crossing if within bounds.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of access interval boundaries to ensure precise computation at the edges of analysis intervals.

* **Tests**
  * Added a new test to verify that interpolation does not occur outside the specified analysis interval, ensuring correct access detection at interval boundaries.

### CODE REVIEW

File: src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp [hunk 1]
```diff
@@ -840,7 +840,7 @@ Array<physics::time::Interval> Generator::computePreciseCrossings(
         Instant intervalStart = anAnalysisInterval.getStart();
 
         // compute start crossing if it is not the start of the requested analysis interval
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp:842]:
```suggestion
        // Compute start crossing if it minus one step is after the analysis interval lower bound
```

File: src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp [hunk 2]
```diff
@@ -859,7 +859,7 @@ Array<physics::time::Interval> Generator::computePreciseCrossings(
         Instant intervalEnd = anAnalysisInterval.getEnd();
 
         // compute end crossing if it is not the end of the requested analysis interval
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp:861]:
```suggestion
        // Compute end crossing if it plus one step is before the analysis interval upper bound
```

File: test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp [hunk 3]
```diff
@@ -1103,6 +1105,68 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Access_Generator, ComputeAccesses_5)
     }
 }
 
+TEST_F(OpenSpaceToolkit_Astrodynamics_Access_Generator, ComputeAccesses_6)
+{
+    // Ensure no interpolation is performed outside of the analysis interval
+
+    {
+        const Instant instant = Instant::J2000();
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp:1113]: Nit: clearer for the reader of the unit test to either put this instant in isoformat, or put the below `interval` instants in `instant + Duration` and `instant - Duration` format

File: test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp [hunk 4]
```diff
@@ -1103,6 +1105,68 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Access_Generator, ComputeAccesses_5)
     }
 }
 
+TEST_F(OpenSpaceToolkit_Astrodynamics_Access_Generator, ComputeAccesses_6)
+{
+    // Ensure no interpolation is performed outside of the analysis interval
+
+    {
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp:1112]: This test only tests exercise the upper bound code path? Or does it exercise both the upper and lower?

If its not much work (idk if you have to "know" what the accesses look like ahead of time to trigger the lower bound condition) it would be good to exercise the other code path as well

**Review by apaletta3 — APPROVED** [review]

File: src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp (hunk 1 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Access/Generator.cpp:842]: Will update it but I think we can make it a bit clearer.

File: test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp (hunk 3 above)
Comment (vishwa2710) [reply test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp:1113]: You want me to instantiate the J2000 epoch as an isoformat for the orbit epoch? It's not even used relative to the access interval, not sure why it matters in this case? But I advocate against instantiating the J2000 as ISOformat for sure, will lead to significant human error.

File: test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp (hunk 4 above)
Comment (vishwa2710) [reply test/OpenSpaceToolkit/Astrodynamics/Access/Generator.test.cpp:1112]: Hashed this out in person and we agree that only the upper bound edge case is hit.
